hack/demo.sh

```diff
@@ -216,20 +207,24 @@ nodes:
     # this directory and any configured providers need to exist during Kubelet's startup
     - hostPath: ${DEMODATA_DIR}/image-credential-provider/
       containerPath: /etc/kubernetes/image-credential-provider/
+    - hostPath: ${DEMODATA_DIR}/etc/kubernetes/image-credentials.json
+      containerPath: /etc/kubernetes/image-credentials.json
+    - hostPath: ${DEMODATA_DIR}/etc/kubernetes/docker-image-credentials.json
+      containerPath: /etc/kubernetes/docker-image-credentials.json
 EOF
 
 kind delete clusters image-credential-provider-test || true
 kind create cluster --config="${DEMODATA_DIR}/kind-config.yaml" --name image-credential-provider-test
 
 docker image pull nginx:latest
-docker image tag docker.io/library/nginx:latest "localhost:${REGISTRY_PORT}/library/nginx:latest"
+docker image tag docker.io/library/nginx:latest "0.0.0.0:${REGISTRY_PORT}/library/nginx:latest"
 
-echo "${REGISTRY_PASSWORD}" | docker login -u "${REGISTRY_USERNAME}" --password-stdin "localhost:${REGISTRY_PORT}"
-docker image push "localhost:${REGISTRY_PORT}/library/nginx:latest"
+echo "${REGISTRY_PASSWORD}" | docker login -u "${REGISTRY_USERNAME}" --password-stdin "0.0.0.0:${REGISTRY_PORT}"
```

Comment:
This change breaks on Linux, no IP SAN for `0.0.0.0` and would require setting up the CA cert properly. I'm surprised this works on macos? Why doesn't localhost work here? The port for the registry is exposed specifically so I'm surprised this doesn't just work fine on macos with localhost?